share/eds/byzantine/bad_encoding.go

```diff
@@ -208,15 +212,15 @@ func (p *BadEncodingProof) Validate(hdr *header.ExtendedHeader) error {
 		log.Debugw("failed to decode shares at height",
 			"height", hdr.Height(), "err", err,
 		)
-		return nil
+		return fmt.Errorf("failed to decode shares: %w", err)
```

Comment:
Previously error was not returned in this case and following ones. Fixed in advance, but I might be missing the idea of how those are handled by function caller.

Reply:
So, the fact that error exist proves the correctness of  BEFP, so its indeed Valid and no error should be returned here. You may add a little comment explaining that here.